controllers/storagecluster/cephfilesystem.go

```diff
@@ -121,11 +122,33 @@ func (obj *ocsCephFilesystems) ensureCreated(r *StorageClusterReconciler, instan
 				return reconcile.Result{}, err
 			}
 		}
+		// create default csi subvolumegroup for the filesystem
+		err = r.createDefaultCsiSubvolumeGroup(cephFilesystem.Name, cephFilesystem.Namespace)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
 	}
 
 	return reconcile.Result{}, nil
 }
 
+func (r *StorageClusterReconciler) createDefaultCsiSubvolumeGroup(filesystemName, filesystemNamespace string) error {
+	objectMeta := metav1.ObjectMeta{Name: "csi", Namespace: filesystemNamespace}
+	cephFilesystemSubVolumeGroup := &cephv1.CephFilesystemSubVolumeGroup{ObjectMeta: objectMeta}
+	mutateFn := func() error {
+		cephFilesystemSubVolumeGroup.Spec = cephv1.CephFilesystemSubVolumeGroupSpec{
+			FilesystemName: filesystemName,
+		}
+		return nil
+	}
+	_, err := ctrl.CreateOrUpdate(r.ctx, r.Client, cephFilesystemSubVolumeGroup, mutateFn)
```

Comment:
no owner ref for this one? and also there should be a check for DeletionTimestamp to avoid updating when its under deletion

Reply:
I think it might make sense to move this CR creation to a separate set of `ensureCreated` and `ensureDeleted` functions under a new file names `cephsubvolumegroup.go` and then call it in the reconcile loop accordingly.
cc @malayparida2000 @iamniting

Reply:
Actually it need all the filesystem name, So we need to query each thing again. And it should be created only after the filesystem get created succefully

Reply:
Yeah, regarding that.
Looking at the code, it doesn't seem like we are creating multiple cephfilsystems, or are we?

Reply:
For internal mode we may do

Reply:
we create only one filesystem as of today

Reply:
Right, so we should consider that as well while implementing this CR creation.

Reply:
Will update it added a TODO once the rook bug is fixed